awx/api/fields.py

```diff
@@ -101,8 +101,15 @@ def __init__(self, **kwargs):
         super(DeprecatedCredentialField, self).__init__(**kwargs)
 
     def to_internal_value(self, pk):
+        if isinstance(pk, str):
+            self.fail('invalid')
```

Comment:
I'm okay with removing these 2 lines, so that it brings its behavior in line with how the other fields work.

Reply:
It's been removed, and now any integer wrapped in quotes gets converted from a string to integer.